tests/platform_helper/utils/test_application.py

```diff
@@ -135,26 +135,36 @@ def test_load_application_does_not_fail_when_addons_params_are_present(
         mock_session = MagicMock(name="session-mock")
         mock_session.client.return_value = mock_client
         mock_client.get_caller_identity.return_value = {"Account": "111111111"}
-        mock_client.get_parameters_by_path.return_value = {
-            "Parameters": [
-                {
-                    "Name": f"/copilot/applications/test/environments/one",
-                    "Value": json.dumps({"name": "one", "accountID": "111111111"}),
-                },
-                {
-                    "Name": f"/copilot/applications/test/environments/two/addons",
-                    "Value": json.dumps(
-                        {"demodjango-redis": {"type": "redis", "environments": {}}}
-                    ),
-                },
-                {
-                    "Name": f"/copilot/applications/test/environments/two/something/else",
-                    "Value": json.dumps(
-                        {"demodjango-redis": {"type": "redis", "environments": {}}}
-                    ),
-                },
-            ]
-        }
+        mock_client.get_parameters_by_path.side_effect = [
```

Comment:
Can you explain the change from `return_value` to `side_effect`?

Reply:
I need the same function to be called twice and return different values on each occasion. I don't _think_ I can achieve that with `return_value`

Reply:
Thanks. I believe you are correct.